libnopegl/src/node_renderother.c

```diff
@@ -104,6 +104,7 @@ struct render_common {
     int topology;
     const struct geometry *geometry;
     struct darray pipeline_descs;
+    struct ngl_node *extra_draw_node;
```

Comment:
Can we have the same mechanism as in `pass.c` ?

Reply:
You mean you'd prefer to have a `darray` here as well?

Reply:
Yup + the automatic resources registrations.

Reply:
Sounds overkill but OK. With regards to automation it cannot be done, renderother doesn't work that way: the resources are immediate parameters, not stored in a generic node dict parameters.

Reply:
Here is the diff with the `darray` thingy:

```diff
diff --git a/libnopegl/src/node_renderother.c b/libnopegl/src/node_renderother.c
index dcea72135..708045d19 100644
--- a/libnopegl/src/node_renderother.c
+++ b/libnopegl/src/node_renderother.c
@@ -108,7 +108,7 @@ struct render_common {
     int topology;
     const struct geometry *geometry;
     struct darray pipeline_descs;
-    struct ngl_node *extra_draw_node;
+    struct darray draw_resources;
 };

 struct rendercolor_opts {
@@ -594,7 +594,11 @@ static int renderhistogram_init(struct ngl_node *node)
 {
     struct renderhistogram_priv *s = node->priv_data;
     struct renderhistogram_opts *o = node->opts;
-    s->common.extra_draw_node = o->stats;
+
+    ngli_darray_init(&s->common.draw_resources, sizeof(struct ngl_node *), 0);
+    if (!ngli_darray_push(&s->common.draw_resources, &o->stats))
+        return NGL_ERROR_MEMORY;
+
     return init(node, &s->common, &o->common, "source_histogram", source_histogram_frag);
 }

@@ -609,7 +613,11 @@ static int renderwaveform_init(struct ngl_node *node)
 {
     struct renderwaveform_priv *s = node->priv_data;
     struct renderwaveform_opts *o = node->opts;
-    s->common.extra_draw_node = o->stats;
+
+    ngli_darray_init(&s->common.draw_resources, sizeof(struct ngl_node *), 0);
+    if (!ngli_darray_push(&s->common.draw_resources, &o->stats))
+        return NGL_ERROR_MEMORY;
+
     return init(node, &s->common, &o->common, "source_waveform", source_waveform_frag);
 }

@@ -1096,8 +1104,9 @@ static int renderwaveform_prepare(struct ngl_node *node)

 static void renderother_draw(struct ngl_node *node, struct render_common *s, const struct render_common_opts *o)
 {
-    if (s->extra_draw_node)
-        ngli_node_draw(s->extra_draw_node);
+    struct ngl_node **draw_resources = ngli_darray_data(&s->draw_resources);
+    for (int i = 0; i < ngli_darray_count(&s->draw_resources); i++)
+        ngli_node_draw(draw_resources[i]);

     struct ngl_ctx *ctx = node->ctx;
     struct pipeline_desc *descs = ngli_darray_data(&s->pipeline_descs);
@@ -1155,6 +1164,7 @@ static void renderother_uninit(struct ngl_node *node, struct render_common *s)
     ngli_darray_reset(&s->pipeline_descs);
     ngli_buffer_freep(&s->vertices);
     ngli_buffer_freep(&s->uvcoords);
+    ngli_darray_reset(&s->draw_resources);
 }

 #define DECLARE_RENDEROTHER(type, cls_id, cls_name) \
```
Do you confirm you prefer this? I'm not a fan of the init done only in some cases vs the uninit always done, but doing this differently will require unrolling a lot of code.

Reply:
Applied.